absorption.py

```diff
@@ -0,0 +1,88 @@
+import scipp as sc
+import mantid.simpleapi as simpleapi
+
+def ensure_sample_in_ws(ws):
+    """
+    test for shape in sample - if not there, add a default vanadium cylinder
+
+    Parameters
+    ----------
+        ws: Mantid Workspace
+            Workspace with Sample defined
+    Returns
+    -------
+    Mantid workspace containing the original sample or a default cylinder sample.
+
+    """
+    sample = ws.sample()   
+    shape_xml = sample.getShape().getShapeXML()
+    import xml.etree.ElementTree as etree
+    root = etree.ElementTree(etree.fromstring(shape_xml)).getroot()
+    # if no sample defined, the content of the shapexml element is empty.
+    if root.text == '  ':
+        shape = '''<cylinder id="cyl">
+        <centre-of-bottom-base x="0.0" y="0.0" z="0.0" />
+        <axis x="0.0" y="0.2" z="0.0"/>
+        <radius val="1"/>
+        <height val="10"/>
+        </cylinder>'''
+        simpleapi.CreateSampleShape(ws, shape)
+        simpleapi.SetSampleMaterial(ws, ChemicalFormula="V")
+    return ws
+
+def absorption_correction(input_variable, **kwargs):
+    """
+    This method is a straightforward wrapper exposing AbsorptionCorrection through scipp
+
+    AbsorptionCorrection calculates an approximation of the attenuation due to absorption
+    and single scattering in a generic sample shape.
+
+    Requirements:
+    - The input workspace must have units of wavelength
+    - The instrument associated with the workspace must be fully defined.
+      (This being a WISH-centric implementation is done with the predefined instr file)
+
+    Parameters
+    ----------
+    input_variable: Scipp dataset with sample defined.
+
+    kwargs: additional arguments to be passed to Mantid's CylinderAbsorption method.
+
+    Returns
+    -------
+    Scipp dataset containing absorption correction in Wavelength units.
+
+    """
+
+    # WISH specific binning
+    lambda_binning = (0.7, 10.35, 5615)
+
+    # Extract sample information
+    sample = input_variable.attrs['sample'].value
+
+    # Create empty workspace with the WISH instrument definitions
+    ws = simpleapi.LoadEmptyInstrument(Filename="WISH_Definition.xml")
+
+    # Assign the unit to X.
+    # Notice that assigning the unit to X is more complex than to Y:
+    # for Y we would have `ws.setYUnit()`
+    ws.getAxis(0).setUnit('Wavelength')
+
+    # Rebin the resulting correction based on default WISH binning
+    lambda_min, lambda_max, number_bins = lambda_binning
+    ws = simpleapi.Rebin(ws, params=[lambda_min, number_bins, lambda_max])
+
+    # Set sample on the workspace
+    ws.setSample(sample)
+
+    # Make sure a sample is defined
+    ws = ensure_sample_in_ws(ws)

```

Comment:
Is this useful? Should you rather just fail here, and let the user of `absorption_correction()` set a sample?

Reply:
I added this in order for the script to run for smaller test files without sample defined.
I don't know if all data files should contain sample? If this is indeed a prerequisite then you're right - we need to throw and inform the user.

Reply:
Well, in practice I think almost all files contain a sample. Furthermore, the Mantid algorithm provides special parameter to define an optional different sample, if required. Overall, having a hard-coded sample does not appear useful (or will even turn out to be a reason for bugs).

If the data array has no sample, you can just skip setting it, and let Mantid handle the failure, maybe?

Reply:
The script is now raising AttributeError on sample-less workspace.

Reply:
Yes, that is what I meant in my last sentence:
> If the data array has no sample, you can just skip setting it, and let Mantid handle the failure, maybe?

Something like
```python
if 'sample' in data.attrs:
  ws.setSample(...)
```

Reply:
Why would we want to rely on Mantid here if we can catch the failure earlier and inform the user? I would suggest
`
if 'sample' not in data.attrs:
    raise AttributeError("Data array contains no sample.")
`

Reply:
Because, as far as I understand, if you specify certain arguments such as `CylinderSampleHeight` to the Mantid algorithm it will not require the sample.

Reply:
I see. Fair point.